source/configuration_manual/fts/flatcurve.rst

```diff
@@ -0,0 +1,168 @@
+.. _fts_backend_flatcurve:
+
+Flatcurve FTS Engine
+====================
+
+.. versionadded:: 2.4.0;3.0.0
+
+Requires Xapian 1.4+ to work properly.
+
+This is a Dovecot FTS plugin to enable message indexing using the
+[Xapian](https://xapian.org/) Open Source Search Engine Library.
+
+The plugin relies on Dovecot to do the necessary stemming. It is intended
+to act as a simple interface to the Xapian storage/search query
+functionality.
+
+This driver supports match scoring and substring matches (on by default),
+which means it is RFC 3501 (IMAP4rev1) compliant. This driver does not
+support fuzzy searches.
+
+The driver passes all of the ImapTest search tests.
+`<https://imapwiki.org/ImapTest>`_
+
+Libraries
+---------
+
+* Xapian: Get ``xapian-core`` v1.4.18 at `<https://xapian.org/download/>`_
+
+Optional parameters
+-------------------
+for the `fts_flatcurve` plugin setting:
+
+ - `commit_limit` - Commit database changes after this many documents are
+                    updated. Higher commit limits will result in faster
+                    indexing for large transactions (i.e. indexing a large
+                    mailbox) at the expense of high memory usage. The default
+                    value should be sufficient to allow indexing in a 256 MB
+                    maximum size process. (integer, set to 0 to use the
+                    Xapian default; DEFAULT: 500)
+ - `max_term_size` - The maximum number of characters in a term to index.
+                     (integer, maximum 200; DEFAULT: 30)
+ - `min_term_size` - The minimum number of characters in a term to index.
+                     (integer; DEFAULT: 2)
+ - `optimize_limit` - Once the database reaches this number of shards,
+                      automatically optimize the DB at shutdown. (integer,
+                      set to 0 to disable; DEFAULT: 10)
+ - `rotate_size` - When the mail ("current") database reaches this number
+                   of messages, it is rotated to a read-only database and
+                   replaced by a new write DB. Most people should not
+                   change this setting. (integer, set to 0 to disable
+                   rotation; DEFAULT: 5000)
+ - `rotate_time` - When the mail ("current") database exceeds this length
+                   of time (in msecs) to commit changes, it is rotated to a
+                   read-only database and replaced by a new write DB. Most
+                   people should not change this setting. (integer, set to 0
+                   to disable rotation; DEFAULT: 5000)
+ - `substring_search` - If enabled, allows substring searches (RFC 3501
+                        compliant). However, this requires significant
+                        additional storage space, so substring searches can
+                        be disabled, if necessary. ("yes" or "no"; DEFAULT:
+                        "yes")
```

Comment:
This is all outdated. These are all individual plugin settings now. See https://github.com/slusarz/dovecot-fts-flatcurve#fts_flatcurve_commit_limit

So this needs to be refactored to use the new dovecot-plugin directive for each setting. And that means settings need to be broken out into separate page (under /settings/plugin/fts-flatcurve-plugin/) from discussion about the beahvior/setup of the fts plugin (this page).  See fts-solr documentation as an example.